pandas/tests/indexing/test_indexing.py

```diff
@@ -231,7 +231,9 @@ def test_setitem_dtype_upcast(self):
 
         df.loc[0, "c"] = "foo"
         expected = DataFrame([{"a": 1, "c": "foo"}, {"a": 3, "b": 2, "c": np.nan}])
-        tm.assert_frame_equal(df, expected)
+        # dataframe dict key order is preserved, so these two
+        # sequences of ops lead to different column order
+        tm.assert_frame_equal(df, expected, check_like=True)
```

Comment:
```suggestion
        tm.assert_frame_equal(df, expected, check_like=not PY36)
```

and then you can remove the comment above